poulpe-service/src/main/java/org/jtalks/poulpe/logic/PermissionManager.java

```diff
@@ -117,6 +118,10 @@ public void changeRestrictions(Entity entity, PermissionChanges changes) {
      * @return {@link PermissionsMap} for {@link Group}
      */
     public GroupsPermissions<ProfilePermission> getPermissionsMapFor(List<Group> groups) {
+        if(isFirst){
```

Comment:
First of all, this condition is not thread safe. Secondary, what if more than one invalid authentication will be stored in cache?